app/org/sagebionetworks/bridge/cache/CacheProvider.java

```diff
@@ -127,24 +114,92 @@ public void setUserSession(final UserSession session) {
             promptToStartRedisIfLocal(e);
             throw new BridgeServiceException(e);
         }
-    }
 
-    public UserSession getUserSession(final String sessionToken) {
+    }
+    
+    public UserSession getUserSession(String sessionToken) {
+        UserSession session = getUserSessionNewVersion(sessionToken);
+        if (session == null) {
+            session = getUserSessionOldVersion(sessionToken);
+        }
+        return session;
+    }
+    
+    private UserSession getUserSessionNewVersion(String sessionToken) {
+        checkNotNull(sessionToken);
+        try {
```

Comment:
Do we need unit tests for the new get methods? Should include the following cases:

* session token not found
* user ID has no session
* session ID mismatch
* session successfully retrieved

Reply:
I added tests for these specifically against the new getSession method alone. Will adjust once we cleanup the old methods.